pepper-apis/src/main/java/org/broadinstitute/ddp/db/dto/EventConfigurationDto.java

```diff
@@ -1,80 +1,150 @@
 package org.broadinstitute.ddp.db.dto;
 
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.commons.lang3.tuple.ImmutablePair;
+import org.apache.commons.lang3.tuple.Pair;
 import org.broadinstitute.ddp.model.activity.types.EventActionType;
 import org.broadinstitute.ddp.model.activity.types.EventTriggerType;
+import org.broadinstitute.ddp.model.activity.types.InstanceStatusType;
 import org.broadinstitute.ddp.model.event.CopyAnswerTarget;
-import org.jdbi.v3.core.mapper.reflect.JdbiConstructor;
-import org.jdbi.v3.sqlobject.customizer.Bind;
-
+import org.broadinstitute.ddp.model.event.NotificationServiceType;
+import org.broadinstitute.ddp.model.event.NotificationType;
 
+/**
+ * A DTO representing the event configuration, left joined with possible trigger and action configuration info
+ */
 public class EventConfigurationDto {
-    private EventTriggerType triggerType;
-    private Integer secondsToWaitBeforePosting;
     private long eventConfigurationId;
+    private EventTriggerType eventTriggerType;
+    private EventActionType eventActionType;
+    private Integer postDelaySeconds;
+    private boolean dispatchToHousekeeping;
     private String preconditionExpression;
     private String cancelExpression;
-    private long activityIdToCreate;
-    private Integer maxInstancesPerUser;
     private Integer maxOccurrencesPerUser;
-    private EventActionType eventActionType;
-    private long announcementMsgTemplateId;
+    private String gcpTopic; // FIXME should not be a base field of EventAction
+
+    /**
+     * Triggers
+     **/
+    /* ACTIVITY_STATUS */
+    private InstanceStatusType instanceStatusType;
+    private Long activityStatusTriggerStudyActivityId;
+
+    /* WORKFLOW_STATE */
+    private Long workflowStateId;
+    private Boolean triggerAutomatically;
+
+    /* JOIN_MAILING_LIST */
+    // No sub-table
+
+    /* DSM_NOTIFICATION */
+    private String dsmNotificationEventType; // FIXME Turn into real enum
+
+    /* MEDICAL_UPDATE */
+    // No sub-table
+
+    /* USER_NOT_IN_STUDY */
+    // No sub-table
+
+    /* USER_REGISTERED */
+    // No sub-table
+
+    /* EXIT_REQUEST */
+    // No sub-table
+
+    /**
+     * Actions
+     **/
+    /* ANNOUNCEMENT */
+    private Long messageTemplateId;
+
+    /* NOTIFICATION */
+    private NotificationType notificationType;
+    private NotificationServiceType notificationServiceType;
+    private Long notificationTemplateId;
+    private Long linkedActivityId;
+
+    /* PDF_GENERATION */
+    // A list of Tuple<pdfDocumentConfigurationId, generateIfMissing>
+    private List<Pair<Long, Boolean>> pdfConfigs = new ArrayList<>();
```

Comment:
There is the `PdfAttachment` class. Have we considered using that instead of a tuple?